app/code/local/Magmodules/Webwinkelconnect/Helper/Data.php

```diff
@@ -97,7 +97,133 @@ public function isProductReviewInviteEnabled(): bool {
             return false;
         }
 
+        $rating_options_collection = Mage::getModel('rating/rating')->getCollection();
+        if ($rating_options_collection->getSize() < 1) {
+            Mage::throwException($this->__('You have no ratings created. Please create at least one rating to use the product reviews feature.'));
+        }
+
+        return true;
+    }
+
+    public function hasCorrectCredentials(?string $webshop_id, ?string $api_key): bool {
+        if (!trim($webshop_id) || !trim($api_key)) {
+            return false;
+        }
+        if ($webshop_id != $this->getShopId() || !hash_equals($api_key, $this->getApiKey())) {
+            return false;
+        }
         return true;
     }
 
+    public function syncProductReview(array $product_review): array {
+        $connection = Mage::getSingleton('core/resource')->getConnection('core_write');
+        if (!$this->isProductReviewInviteEnabled()) {
+            throw new RuntimeException("Product reviews are disabled");
+        }
+        try {
+            $connection->beginTransaction();
+            $product_id = $product_review['product_review']['product_id'];
+            $parent_id = Mage::getModel('catalog/product_type_configurable')->getParentIdsByChild($product_id);
+            if (!empty($parent_id)) {
+                $product_id = $parent_id[0];
+            }
+
+            ($review = Mage::getModel('review/review'))
+                ->setEntityPkValue($product_id)
+                ->setStatusId(Mage_Review_Model_Review::STATUS_APPROVED)
+                ->setTitle($product_review['product_review']['title'])
+                ->setDetail($product_review['product_review']['review'])
+                ->setEntityId($review->getEntityIdByCode(Mage_Review_Model_Review::ENTITY_PRODUCT_CODE))
+                ->setStoreId(Mage::app()->getStore()->getStoreId())
+                ->setStores([Mage::app()->getStore()->getStoreId()])
+                ->setCustomerId($this->getCustomerId($product_review['product_review']['reviewer']['email']))
+                ->setNickname($product_review['product_review']['reviewer']['name'])
+                ->save();
+
+            $rating_option_id = Mage::getModel('rating/rating_option')->getCollection()
```

Comment:
This is already handled in the isProductReviewInviteEnabled method. It throws an exception if there are no rating option IDs.

Reply:
What if you only have options 1-3 configured, and you try to sync a rating with score 5?

Reply:
This would only be possible if somebody tampered with the DB or is using a different module for ratings, but I see your point and will handle the possibility.